### PR TITLE
Add ScANN example

### PR DESCRIPTION
**Pull request by abheesht17**
Omitted

- Accuracy computation
- Tuning ScANN

Should we include these?

My question is: if we want to include these, we might have to manually iterate over the test dataset and compute predictions. Might be a little tedious. Let me know if we want to do this, @hertschuh.

### CODE REVIEW

File: examples/scann.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+Note: ScANN does not have its own separate layer in KerasRS because the ScANN
+library is TensorFlow-only. Here, in this example, we directly use the ScANN
+library and demonstrate its usage with KerasRS.
+
+## Imports
+
+Let's install the `scann` library and import all necessary packages.
+"""
+
+# ruff: noqa: E402
```
Comment (hertschuh) [inline examples/scann.py:37]: Do you need this? The imports look like they're at the top.

File: examples/scann.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+[ScANN](https://research.google/blog/announcing-scann-efficient-vector-similarity-search/),
+developed by Google Research, is a high-performance library designed for
+dense vector similarity search at scale. It efficiently indexes a database of
+candidate embeddings, enabling rapid search during inference. By leveraging
+advanced vector compression techniques and finely tuned algorithms, ScaNN
+strikes an optimal balance between speed and accuracy. As a result, it can
+significantly outperform brute-force search methods, delivering fast retrieval
+with minimal loss in accuracy.
+
+We will start with copying over code from the
```
Comment (hertschuh) [inline examples/scann.py:23]: Nitpick: instead of saying "We will start with copying over code from the", say "We will start with the same code as the".

The former makes it sound like the code is running a `cp` command or something.

File: examples/scann.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+        candidate_embeddings
+        * keras.random.uniform(keras.ops.shape(candidate_embeddings))
+        for _ in range(100)
+    ],
+    axis=0,
+)
+
+user_embedding = model.user_embedding(keras.ops.array([10, 5, 42, 345]))
+
+print(f"{candidate_embeddings.shape=}", f"{user_embedding.shape=}")
```
Comment (hertschuh) [inline examples/scann.py:182]: Is it intentional that this `print` is here?

**Comment by hertschuh** [comment]
> Omitted
> 
> * Accuracy computation
> * Tuning ScANN
> 
> Should we include these?
> 
> My question is: if we want to include these, we might have to manually iterate over the test dataset and compute predictions. Might be a little tedious. Let me know if we want to do this, @hertschuh.

I don't think we need these.

File: examples/scann.py (hunk 1 above)
Comment (abheesht17) [reply examples/scann.py:37]: For some reason, the `"""shell` block isn't considered a string by ruff, and it keeps throwing E402.

File: examples/scann.py (hunk 3 above)
Comment (abheesht17) [reply examples/scann.py:182]: Oh, sorry. I was using it for debugging

**Review by hertschuh — APPROVED** [review]
Thanks!
